src/lib/db/transactional.ts

```diff
@@ -0,0 +1,36 @@
+import { Knex } from 'knex';
+import { Transactional } from 'lib/types/stores/transactional';
+
+export abstract class Transactor<T> implements Transactional<T> {
+    args: any[];
+
+    // eslint-disable-next-line
+    constructor(...args: any) {
+        this.args = args;
+    }
+
+    transactional(transaction: Knex.Transaction): T {
```

Comment:
Would it make sense to rename this as `join` or `joinTransaction`?

Reply:
`join` seems dangerous - I would at least start wondering how on earth a SQL join works in this context. I could go for `joinTransaction` but I kinda want something that expresses that you're getting a new object here too, `transactional` might not be it

Reply:
Then transactional might work well, although `cloneWithTransaction` could work for this at the risk of being too verbose and explicit

Reply:
Yeah `cloneWithTransaction` isn't bad, I think the tricky thing with this name is that I'm both trying to mask the fact that I'm doing a clone in the service layer so that you can ignore that implementation detail while still trying to be expressive about what details you need to be aware of in the calling code (the stuff behind the scenes will fall off the stack when the closing block completes).

I'm open to going with `cloneWithTransaction` but let's let a few more folks weigh in on this one

Reply:
I agree with you, I'd like to hide the fact that you're cloning, as I think it's an implementation detail